### ISSUE TITLE
Lutris deleted my entire games folder, add protections

### ISSUE THREAD

**Issue by DEFCON001** [lutris/lutris#4330]
This is mostly my fault for stupidity, but should have proper protections in place so it doesn't happen to non-technical users.

How to replicate:
- Have a games folder with locally installed Lutris games on an external Hard Drive
- Go to install a game
- Change install location from OS drive to hard drive games directory
- Realise you want to get the original directory name for said game install
- Press cancel and tick delete installed files (thinking this will remove files from the default drive for some reason)
- Games folder nuked and not able to be saved unless you backed up your saves (luckily I think I have backups of my favourite games save files).

Please implement a warning or something if the canceling of an installation will erase more than 1 game lutris is aware of, or make some way of reversing changes (although I don't think this is feasible tbh).

**Comment by DEFCON001** [comment]
Update: I had my most important games backed up, so all is fine, but there should still be a protection against deleting all your lutris games this way.

**Comment by strycore** [comment]
we're eventually going to change how files are deleted.
I've added several checks over the years and none of them are sufficient. The only safe way to do it is referencing the files created during an install and only delete those

**Comment by MathiasMell** [comment]
Same happend to me two hours ago..is there any chance to recover the Games folder? 
I tried testdisk but it won't show the folder at all?

**Comment by DEFCON001** [comment]
Sadly if you didn't back it up or use something like btrfs, it will almost certainly be gone.

**Comment by MathiasMell** [comment]
I was able to recover some files with r-linux, but ended up rebuilding the whole configuration.
Lesson learned ... using Time Shift and rsync now.. :)

**Comment by danieljohnson2** [comment]
Lutris already has some protections against naughty deletions, but it does not use them when cancelling an installation, but only when removing an installed game. That can't be right.

I've added checks when removing the files during cancellation of installation. Also, I've added check to protect the system-wide default game directory, in case that is not set to '~/Games' (which was protected before, at least for game-removal).

That should handle your particular case, at the least.
